### PR TITLE
ref: Migrate to `isTeamPlan` GQ field 

### PR DESCRIPTION
**Pull request by RulaKhaled**
# Description
This PR aims to remove the isTeamPlan helper function in Gazebo in favor of the new GraphQL field for the same.

fyi we're gonna come back once available plans is refactored to pass down isTeamPlan instead of checking for plans enum for the selected plans, this is just a weird edge case.

# Notable Changes
- updated tests 
- removed helper
- updated the plan 

# Screenshots
nothing visual changed 

issue: [#2998](https://github.com/codecov/engineering-team/issues/2998)

# Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. In 2022 this entity acquired Codecov and as result Sentry is going to need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm/UpdateBlurb/UpdateBlurb.tsx [hunk 1]
```diff
@@ -48,7 +49,7 @@ const UpdateBlurb = ({
       {diffSeats && (
         <li className="pl-2">{`You are changing seats from ${currentPlan?.planUserCount} to [${seats}]`}</li>
       )}
-      {diffBillingType && (
+      {diffBillingType && !currentIsFree && (
```
Comment (ajay-sentry) [inline src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm/UpdateBlurb/UpdateBlurb.tsx:52]: weird this is showing up, I thought I had committed it already

**Review by ajay-sentry — APPROVED** [review]

File: src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm/UpdateBlurb/UpdateBlurb.tsx (hunk 1 above)
Comment (RulaKhaled) [reply src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm/UpdateBlurb/UpdateBlurb.tsx:52]: yah ikr?
